beacon/validator/src/test/java/tech/pegasys/teku/validator/coordinator/DepositProviderTest.java

```diff
@@ -164,18 +174,47 @@ void numberOfDepositsThatCanBeIncludedMoreThanMaxDeposits() {
     checkThatDepositProofIsValid(deposits);
   }
 
+  @Test
+  void noDepositsIncludedIfFormerDepositMechanismHasBeenDisabled() {
+    setup(16, SpecMilestone.ELECTRA);
+    updateStateEth1DepositIndex(5);
+    updateStateDepositReceiptsStartIndex(5);
+
+    final SszList<Deposit> deposits = depositProvider.getDeposits(state, randomEth1Data);
+
+    assertThat(deposits).isEmpty();
+  }
+
+  @Test
+  void getsRemainingEth1PendingDepositsIfElectraIsEnabled() {
+    setup(16, SpecMilestone.ELECTRA);
+    updateStateEth1DepositIndex(5);
+    updateStateEth1DataDepositCount(20);
+    // 16th deposit is using the new mechanism
+    updateStateDepositReceiptsStartIndex(16);
+
+    mockDepositsFromEth1Block(0, 10);
+    mockDepositsFromEth1Block(10, 20);
+
+    final SszList<Deposit> deposits = depositProvider.getDeposits(state, randomEth1Data);
+
+    // the pending Eth1 deposits (deposit_receipt_start_index - eth1_deposit_index)
+    assertThat(deposits).hasSize(11);
```

Comment:
5th is already in state, next 6,7,8,9,10,   11, 12,13,14,15
10 total, no?

Reply:
eth1_deposit_index is a bit confusing, but refers to the next deposit index to be processed, so 5th is not in state yet.

Reply:
Got it! Could you please add a comment about it for clarification, how do we get 11